java/src/test/java/com/saucelabs/simplesauce/SauceOptionsTest.java

```diff
@@ -1,31 +1,158 @@
+    }
 
     @Test
-    public void sauceOptions_defaultBrowser_setToChrome() {
-        assertEquals("chrome", options.getBrowserName());
+    public void acceptsChromeOptionsClass() {
+        ChromeOptions chromeOptions = new ChromeOptions();
+        chromeOptions.addArguments("--foo");
+        chromeOptions.setUnhandledPromptBehaviour(DISMISS);
+
+        sauceOptions = new SauceOptions(chromeOptions);
+
+        assertEquals("chrome", sauceOptions.getBrowserName());
+        assertEquals(chromeOptions, sauceOptions.getSeleniumCapabilities());
     }
 
     @Test
-    public void sauceOptions_defaultBrowserVersion_setToLatest() {
-        assertEquals("latest", options.getBrowserVersion());
+    public void acceptsEdgeOptionsClass() {
+        EdgeOptions edgeOptions = new EdgeOptions();
+        edgeOptions.setPageLoadStrategy("eager");
+
+        sauceOptions = new SauceOptions(edgeOptions);
+
+        assertEquals("MicrosoftEdge", sauceOptions.getBrowserName());
+        assertEquals(edgeOptions, sauceOptions.getSeleniumCapabilities());
     }
 
     @Test
-    public void sauceOptions_defaultOS_setToWindows() {
-        assertEquals("Windows 10", options.getOperatingSystem());
+    public void acceptsFirefoxOptionsClass() {
+        FirefoxOptions firefoxOptions = new FirefoxOptions();
+        firefoxOptions.addArguments("--foo");
+        firefoxOptions.addPreference("foo", "bar");
+        firefoxOptions.setUnhandledPromptBehaviour(DISMISS);
+
+        sauceOptions = new SauceOptions(firefoxOptions);
+
+        assertEquals("firefox", sauceOptions.getBrowserName());
+        assertEquals(firefoxOptions, sauceOptions.getSeleniumCapabilities());
     }
 
     @Test
-    public void withChrome_browser_setToChrome() {
-        options.withChrome();
-        assertNotNull(options.getChromeOptions());
+    public void acceptsInternetExplorerOptionsClass() {
+        InternetExplorerOptions internetExplorerOptions = new InternetExplorerOptions();
+        internetExplorerOptions.requireWindowFocus();
+        internetExplorerOptions.setPageLoadStrategy(PageLoadStrategy.EAGER);
+        internetExplorerOptions.setUnhandledPromptBehaviour(DISMISS);
+
+        sauceOptions = new SauceOptions(internetExplorerOptions);
+
+        assertEquals("internet explorer", sauceOptions.getBrowserName());
+        assertEquals(internetExplorerOptions, sauceOptions.getSeleniumCapabilities());
+    }
+
+    @Test
+    public void acceptsSafariOptionsClass() {
+        SafariOptions safariOptions = new SafariOptions();
+        safariOptions.setAutomaticInspection(true);
+        safariOptions.setAutomaticProfiling(true);
+
+        sauceOptions = new SauceOptions(safariOptions);
+
+        assertEquals("safari", sauceOptions.getBrowserName());
+        assertEquals(safariOptions, sauceOptions.getSeleniumCapabilities());
+    }
+
+    @Test
+    @Ignore("Not Implemented Yet")
+    public void setsCapabilitiesFromMap() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4736477)